src/lib.rs

```diff
@@ -105,101 +102,93 @@ impl Plugin for BevyLocalCommandsPlugin {
-    mut active_process_map: ResMut<ActiveProcessMap>,
 ) {
-    for run_shell_command in run_process_event.read() {
-        // Assemble the command
-        let mut cmd = Command::new(run_shell_command.program.clone());
-        cmd.args(run_shell_command.arguments.clone())
-            .stdout(Stdio::piped());
-
-        let command = format!("{cmd:?}");
-
-        let Ok(active_process) = spawn_process(cmd) else {
-            process_error_event.send(ProcessError {
-                command,
-                info: ProcessErrorInfo::FailedToStart,
-            });
-            continue;
-        };
-
-        let pid = active_process.process.id();
-
-        process_started_event.send(ProcessStarted { command, pid });
-
-        active_process_map.0.insert(pid, active_process);
+    for (entity, mut local_command) in query.iter_mut() {
+        match spawn_process(&mut local_command.command) {
+            Ok(process) => {
+                commands.entity(entity).insert(process);
+            },
+            Err(_) => {
+                process_error_event.send(ProcessError {
+                    entity,
+                    info: ProcessErrorInfo::FailedToStart,
+                });
+            },
+        }
     }
 }
 
+/// Periodically empty each processes' output buffer and send the new lines as [`ProcessOutputEvent`].
 fn handle_process_output(
-    mut active_process_map: ResMut<ActiveProcessMap>,
+    query: Query<(Entity, &Process)>,
     mut process_output_event: EventWriter<ProcessOutput>,
 ) {
-    for (&pid, active_process) in active_process_map.0.iter_mut() {
-        if let Ok(mut output_buffer) = active_process.output_buffer.0.lock() {
+    for (entity, process) in query.iter() {
+        if let Ok(mut output_buffer) = process.output_buffer.0.lock() {
             // Send the buffered output in the event while clearing the output buffer
             let mut output = Vec::<String>::new();
             std::mem::swap(&mut *output_buffer, &mut output);
 
             if !output.is_empty() {
-                process_output_event.send(ProcessOutput {
-                    pid,
-                    command: format!("{:?}", active_process.command),
-                    output,
-                });
+                process_output_event.send(ProcessOutput { entity, output });
             }
         }
     }
 }
 
 fn handle_kill_process(
-    mut active_process_map: ResMut<ActiveProcessMap>,
+    mut query: Query<&mut Process>,
     mut kill_process_event: EventReader<KillProcess>,
 ) {
     for kill_shell_command in kill_process_event.read() {
-        let pid = kill_shell_command.0;
+        let entity: Entity = kill_shell_command.0;
 
-        if let Some(active_process) = active_process_map.0.get_mut(&pid) {
+        if let Ok(mut process) = query.get_mut(entity) {
             // Kill the process
             // TODO: Handle unwrap
-            active_process.process.kill().unwrap();
+            process.process.kill().unwrap();
```

Comment:
Might as well use your Process.kill() implementation here and shorten it to `process.kill().unwrap()`

Reply:
Same question regarding despawning here - do we want to despawn the entity?

Reply:
Correct, the entity is despawned in the completion handler